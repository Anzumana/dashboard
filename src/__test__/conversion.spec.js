import {
	convertLineString,
	convertSubgraph ,
	convertCoordinatesToPath
} from '../lib/utils';

test('convert Linestring  from backend ( affected_subgraph ) to path array for deck gl', () => {
	var testData = "LINESTRING(9.7441138 52.3725554,9.7440523 52.3720947,9.7440011 52.3717967,9.7439862 52.3716662,9.7440009 52.3715866)";
	expect(convertLineString(testData)).toMatchSnapshot();
});

// we use this second line string in our test for our affected graph conversion 
// so lets make first sure that it converts correctly
test('convert second Linestring  from backend ( affected_subgraph ) to path array for deck gl', () => {
	var testData = "LINESTRING(9.8135175 52.3172347,9.8136214 52.31745,9.8146824 52.3179731,9.8149586 52.3180234)"
	expect(convertLineString(testData)).toMatchSnapshot();
});

test('convert affected_subgraph  from backend to path array for deck gl', () => {
	var testData = ["LINESTRING(9.7441138 52.3725554,9.7440523 52.3720947,9.7440011 52.3717967,9.7439862 52.3716662,9.7440009 52.3715866)",
		"LINESTRING(9.8135175 52.3172347,9.8136214 52.31745,9.8146824 52.3179731,9.8149586 52.3180234)"];
	expect(convertSubgraph(testData)).toMatchSnapshot();
});
test('convert graphmaster coordinates to path layer', () =>{
	var testData = "52.194516500000006 8.5446034|52.194518900000006 8.5437438|52.19448310000001 8.5413435|52.1944481 8.5397423|52.194395500000006 8.5384442|52.1942171 8.5349027|52.193928500000005 8.5311003|52.193589800000005 8.5273184|52.1933695 8.5248936|52.1933202 8.524400100000001|52.1932084 8.523177|52.1930308 8.521240500000001|52.1928105 8.5189713|52.192508000000004 8.515473700000001|52.1922975 8.5126681|52.1921692 8.510570600000001|52.1921101 8.5092885|52.1920311 8.5070194|52.1920081 8.5047663|52.1920081 8.504144100000001|52.1921265 8.5006626|52.1922712 8.498608|52.192492200000004 8.4965954|52.1927694 8.4946377|52.1931919 8.492337000000001|52.193662200000006 8.4901161|52.194579600000004 8.4864844|52.1953265 8.4836452|52.196054000000004 8.4806924|52.1967744 8.477604900000001|52.197020900000005 8.4763534|52.1972589 8.475002|"
expect(convertCoordinatesToPath(testData)).toMatchSnapshot();
});
