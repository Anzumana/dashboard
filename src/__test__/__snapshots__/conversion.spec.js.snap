// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert Linestring  from backend ( affected_subgraph ) to path array for deck gl 1`] = `
Array [
  Array [
    9.7441138,
    52.3725554,
  ],
  Array [
    9.7440523,
    52.3720947,
  ],
  Array [
    9.7440011,
    52.3717967,
  ],
  Array [
    9.7439862,
    52.3716662,
  ],
  Array [
    9.7440009,
    52.3715866,
  ],
]
`;

exports[`convert affected_subgraph  from backend to path array for deck gl 1`] = `
Array [
  Object {
    "path": Array [
      Array [
        9.7441138,
        52.3725554,
      ],
      Array [
        9.7440523,
        52.3720947,
      ],
      Array [
        9.7440011,
        52.3717967,
      ],
      Array [
        9.7439862,
        52.3716662,
      ],
      Array [
        9.7440009,
        52.3715866,
      ],
    ],
  },
  Object {
    "path": Array [
      Array [
        9.8135175,
        52.3172347,
      ],
      Array [
        9.8136214,
        52.31745,
      ],
      Array [
        9.8146824,
        52.3179731,
      ],
      Array [
        9.8149586,
        52.3180234,
      ],
    ],
  },
]
`;

exports[`convert graphmaster coordinates to path layer 1`] = `
Array [
  Array [
    8.5446034,
    52.194516500000006,
  ],
  Array [
    8.5437438,
    52.194518900000006,
  ],
  Array [
    8.5413435,
    52.19448310000001,
  ],
  Array [
    8.5397423,
    52.1944481,
  ],
  Array [
    8.5384442,
    52.194395500000006,
  ],
  Array [
    8.5349027,
    52.1942171,
  ],
  Array [
    8.5311003,
    52.193928500000005,
  ],
  Array [
    8.5273184,
    52.193589800000005,
  ],
  Array [
    8.5248936,
    52.1933695,
  ],
  Array [
    8.524400100000001,
    52.1933202,
  ],
  Array [
    8.523177,
    52.1932084,
  ],
  Array [
    8.521240500000001,
    52.1930308,
  ],
  Array [
    8.5189713,
    52.1928105,
  ],
  Array [
    8.515473700000001,
    52.192508000000004,
  ],
  Array [
    8.5126681,
    52.1922975,
  ],
  Array [
    8.510570600000001,
    52.1921692,
  ],
  Array [
    8.5092885,
    52.1921101,
  ],
  Array [
    8.5070194,
    52.1920311,
  ],
  Array [
    8.5047663,
    52.1920081,
  ],
  Array [
    8.504144100000001,
    52.1920081,
  ],
  Array [
    8.5006626,
    52.1921265,
  ],
  Array [
    8.498608,
    52.1922712,
  ],
  Array [
    8.4965954,
    52.192492200000004,
  ],
  Array [
    8.4946377,
    52.1927694,
  ],
  Array [
    8.492337000000001,
    52.1931919,
  ],
  Array [
    8.4901161,
    52.193662200000006,
  ],
  Array [
    8.4864844,
    52.194579600000004,
  ],
  Array [
    8.4836452,
    52.1953265,
  ],
  Array [
    8.4806924,
    52.196054000000004,
  ],
  Array [
    8.477604900000001,
    52.1967744,
  ],
  Array [
    8.4763534,
    52.197020900000005,
  ],
  Array [
    8.475002,
    52.1972589,
  ],
]
`;

exports[`convert second Linestring  from backend ( affected_subgraph ) to path array for deck gl 1`] = `
Array [
  Array [
    9.8135175,
    52.3172347,
  ],
  Array [
    9.8136214,
    52.31745,
  ],
  Array [
    9.8146824,
    52.3179731,
  ],
  Array [
    9.8149586,
    52.3180234,
  ],
]
`;
